src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1469,8 +1465,7 @@ def enable_features(cmd, client, resource_group_name, cluster_name, features, ku
         cmd_helm_upgrade.extend(["--kube-context", kube_context])
     if enable_azure_rbac:
         cmd_helm_upgrade.extend(["--set", "systemDefaultValues.guard.enabled=true"])
-        cmd_helm_upgrade.extend(["--set", "systemDefaultValues.guard.clientId={}".format(azrbac_client_id)])
-        cmd_helm_upgrade.extend(["--set", "systemDefaultValues.guard.clientSecret={}".format(azrbac_client_secret)])
+        cmd_helm_upgrade.extend(["--set", "systemDefaultValues.guard.authnMode=arc"])  # new mode for latest OBO changes
```

Comment:
you can remove the `# new mode for latest OBO changes` comment and add a better one on top of this flag to explain this change

Reply:
Updated here: https://github.com/Azure/azure-cli-extensions/pull/6675